api/envoy/api/v2/route/route.proto

```diff
@@ -739,7 +739,7 @@ message RouteAction {
     // Indicates that during forwarding, the host header will be swapped with
     // this value.
     //
-    // [#next-major-version: host_rewrite_literal]
+    // [#rename-at-next-major-version: host_rewrite_literal]
```

Comment:
Should we make these first class proto options that live as proto extensions? It's a little cleaner to have everything represented in proto rather than comment annotations that need to be scraped. I think the comment annotations make sense for protodoc, but not other stuff.

Reply:
I was thinking of that and fine either way. The current situation is we only use proto annotation for things used in runtime (i.e. validate, deprecate, previous_message_type). WDYT?

Reply:
+1 for annotations. I think it's more clear and easier to type check.

Reply:
So, we don't need the renames at runtime for reflection upgrading, but we might need them programatically in the Clang tool I'm building, since it needs to know how to perform the renames in the source code. It might be possible to infer them by knowing the previous / new proto descriptors and correlation corresponding field numbers in them though.

In any case, even for type checking improvements alone, I reckon let's do it as a proper extension.

Reply:
OK will revise it soon.